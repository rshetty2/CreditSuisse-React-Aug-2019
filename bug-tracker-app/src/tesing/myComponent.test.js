import MyComponent from './myComponent';
import { shallow } from 'enzyme';
import React from 'react';

it('should render correctly with no props', () => {
  const component = shallow(<MyComponent/>);
  
  expect(component).toMatchSnapshot();
});

it('should change the state when the button is clicked', () =>{
	const component = shallow(<MyComponent/>);
	component
		.find('input#button1')	
		.simulate('click');

	expect(component.state('message')).toEqual('Button clicked');
})


test('should set the given props', () =>{
	const component = shallow(<MyComponent xyz="100"/>);
	expect(component.instance().props.xyz).toEqual('100');
})