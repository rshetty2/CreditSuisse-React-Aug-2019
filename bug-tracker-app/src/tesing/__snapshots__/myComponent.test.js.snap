// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly with no props 1`] = `
<div>
  <div
    id="div1"
  />
  <h1>
    Test Component
  </h1>
  <input
    id="button1"
    onClick={[Function]}
    type="button"
    value="Click"
  />
</div>
`;
